packages/media-query-mixin/media-query-mixin.ts

```diff
@@ -0,0 +1,125 @@
+import { LitElement, property } from 'lit-element';
+import { PropertyDeclaration } from 'lit-element/lib/updating-element.js';
+
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+export function mediaProperty(opts?: MediaDeclaration): (proto: object, name?: PropertyKey) => any {
+  return property(opts as PropertyDeclaration);
+}
+
+export interface MediaDeclaration {
+  readonly media?: string;
+}
+
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+type Constructor<T = object> = new (...args: any[]) => T;
+
+export const MediaQueryMixin = <T extends Constructor<LitElement>>(
+  base: T
+): T & Constructor<LitElement> => {
+  class MediaQuery extends base {
+    private static _mediaQueries: {
+      [prop: string]: { query: MediaQueryList; updating?: boolean };
+    } = {};
+
+    private static _queryProps: Set<string> = new Set();
+
+    private _boundQueries: { [prop: string]: () => void } = {};
+
+    /**
+     * Extend the LitElement `createProperty` method to watch media query.
+     */
+    static createProperty(name: PropertyKey, options: PropertyDeclaration) {
+      const { media } = options as MediaDeclaration;
+      const prop = name as string;
+
+      const isMediaString = typeof media === 'string';
+      const opts = isMediaString
+        ? {
+            type: Boolean,
+            attribute: false,
+            noAccessor: true,
+            media
+          }
+        : options;
+
+      ((base as unknown) as typeof LitElement).createProperty(name, opts);
+
+      if (isMediaString) {
+        const query = window.matchMedia(media as string);
+        this._mediaQueries[prop] = { query };
+        this._queryProps.add(prop);
```

Comment:
Why do we need `_queryProps ` if we can get a reference with `_mediaQueries[prop]`?

Reply:
Removed `_queryProps`.